tool/execution/parallel/README.md

```diff
@@ -0,0 +1,162 @@
+# Parallel Execution
+
+Library for task parallelization and asynchronous execution.
+
+### References
+
+* Module type - [tool](../../docs/architecture.md#tool)
+* Dependency type - [static](../../docs/architecture.md#static_dependencies)
+* Public API
+    * Core - [Parallel.kt](./src/main/kotlin/flank/exection/parallel/Parallel.kt)
+    * Execution - [Execute.kt](./src/main/kotlin/flank/exection/parallel/Execute.kt)
+    * Task factories DSL - [Create.kt](./src/main/kotlin/flank/exection/parallel/Create.kt)
+    * Graph validation - [Create.kt](./src/main/kotlin/flank/exection/parallel/Validate.kt)
+    * Graph reduction - [Reduce.kt](./src/main/kotlin/flank/exection/parallel/Reduce.kt)
+* Example usage - [Example.kt](./src/test/kotlin/flank/exection/parallel/Example.kt)
+
+# Design
+
+Imagine a complicated long-running `execution` that needs to perform several operations (`tasks`) in correct order to
+collect a required `data` and produce `side effects`. Any `execution` like that, can be modeled as set of unrelated
```

Comment:
```suggestion
collect required `data` and produce `side effects`. Any `execution` like that, can be modeled as a set of unrelated
```
data is plural in this case and doesnt need `a` :) And set in this case needs it :'(